### PR TITLE
Generalize pallets

### PR DESCRIPTION
**Pull request by vovac12**

### CODE REVIEW

File: pallets/trustless-eth-bridge/migration-app/Cargo.toml [hunk 1]
```diff
@@ -17,20 +17,20 @@ hex = { package = "rustc-hex", version = "2.1.0", default-features = false }
...
-sp-io = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.28", default-features = false }
-sp-runtime = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.28", default-features = false }
+frame-benchmarking = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.19", default-features = false, optional = true }
+frame-support = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.19", default-features = false }
+frame-system = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.19", default-features = false }
+sp-core = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.19", default-features = false }
+sp-std = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.19", default-features = false }
+sp-io = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.19", default-features = false }
+sp-runtime = { git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.19", default-features = false }
```
Comment (wer1st) [inline pallets/trustless-eth-bridge/migration-app/Cargo.toml]: Why older branch?

File: pallets/trustless-eth-bridge/substrate-app/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+//! ERC20App pallet benchmarking
```
Comment (wer1st) [inline pallets/trustless-eth-bridge/substrate-app/src/benchmarking.rs:31]: Add copyright header please

File: pallets/trustless-eth-bridge/substrate-app/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,395 @@
+//! # ERC20
```
Comment (wer1st) [inline pallets/trustless-eth-bridge/substrate-app/src/lib.rs]: Copyright

File: pallets/trustless-eth-bridge/substrate-app/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+            );
+            Self::deposit_event(Event::Minted(
+                network_id, asset_id, sender, recipient, amount,
+            ));
+            Ok(())
+        }
+
+        /*
+        Common exstrinsics
+         */
```
Comment (wer1st) [inline pallets/trustless-eth-bridge/substrate-app/src/lib.rs]: Minor: this comment style looks odd.
I think it's better to use:
`// regular comment`
or
`/*`
` * doxygen style comment`
` */`

File: pallets/trustless-eth-bridge/substrate-app/src/lib.rs (hunk 3 above)
Comment (wer1st) [reply pallets/trustless-eth-bridge/substrate-app/src/lib.rs]: The same for all other .rs files

File: pallets/trustless-eth-bridge/substrate-channel/src/inbound/mod.rs [hunk 5]
```diff
@@ -0,0 +1,230 @@
+//! Channel for passing messages from ethereum to substrate.
```
Comment (N1ghtStorm) [inline pallets/trustless-eth-bridge/substrate-channel/src/inbound/mod.rs:31]: Maybe it is better to add a copyright header

File: pallets/trustless-eth-bridge/substrate-channel/src/inbound/mod.rs [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    #[pallet::event]
+    //#[pallet::generate_deposit(pub(super) fn deposit_event)]
```
Comment (N1ghtStorm) [inline pallets/trustless-eth-bridge/substrate-channel/src/inbound/mod.rs]: maybe this line should be uncommented?

Comment (vovac12) [reply pallets/trustless-eth-bridge/substrate-channel/src/inbound/mod.rs]: We will have unused warning

**Pull request merged**